tools/update-dll-msvc.bat

```diff
@@ -21,29 +18,106 @@ rem \     'unix' : 'gmake',
 rem \    },
 rem \ }
 
+setlocal enabledelayedexpansion
+
 if "%PROCESSOR_ARCHITECTURE%"=="AMD64" (
     set vimproc_arch=64
-    set cpu_arch=AMD64
-    set msvc_arch=amd64
 ) else (
     set vimproc_arch=32
-    set cpu_arch=x86
-    set msvc_arch=x86
 )
 
 set vimproc_dllname=vimproc_win%vimproc_arch%.dll
 
+if exist "%~dp0vswhere.exe" (
+    REM  found the lasted version of Visual C compiler
+    for /f "usebackq tokens=*" %%i in (`"%~dp0vswhere.exe" -latest -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath`) do (
+        set InstallDir=%%i
+
+        if exist "%InstallDir%\VC\Auxiliary\Build\Microsoft.VCToolsVersion.default.txt" (
+            set /p msvc_ver=<"%InstallDir%\VC\Auxiliary\Build\Microsoft.VCToolsVersion.default.txt"
+
+            rem Trim
+            set msvc_ver=!msvc_ver: =!
+        )
+
+        if exist "%InstallDir%\VC\Auxiliary\Build\vcvars%vimproc_arch%.bat" (
+            call "%InstallDir%\VC\Auxiliary\Build\vcvars%vimproc_arch%.bat"
+        )
+    )
+) else if defined VS140COMNTOOLS (
+    REM  Microsoft Visual Studio 2015 installed
+    set msvc_ver=14
+    call "%VS140COMNTOOLS%\..\..\VC\bin\vcvars%vimproc_arch%.bat"
+) else if defined VS120COMNTOOLS (
+    REM  Microsoft Visual Studio 2013 installed
+    set msvc_ver=12
+    call "%VS120COMNTOOLS%\..\..\VC\bin\vcvars%vimproc_arch%.bat"
+) else if defined VS110COMNTOOLS (
+    REM  Microsoft Visual Studio 2012 installed
+    set msvc_ver=11
+    call "%VS110COMNTOOLS%\..\..\VC\bin\vcvars%vimproc_arch%.bat"
+) else if defined VS100COMNTOOLS (
+    REM  Microsoft Visual Studio 2010 installed
+    set msvc_ver=10
+    call "%VS100COMNTOOLS%\..\..\VC\bin\vcvars%vimproc_arch%.bat"
+) else if defined VS90COMNTOOLS (
```

Comment:
FYI, Vim 8.2 has dropped support for VC 2008 or older.

Reply:
@k-takata , so for your opinion, I should remove to support VC2008 or older, that will be only can use vswhere.exe to detect the all the version which is VC 2010 and newer. For batch file, it will more easy to implement it to support the different version of Visual Studio. By the way, what the minimal version of Visual C compiler can be sucessful compiled for vimproc.vim plugin?

Reply:
Currently we use Windows SDK 7.1 (= VC 2010) for building vimproc on CI.
VC 2008 or older may work, but it is not tested.

Reply:
@k-takata , I have just push the new commit to improve the search method for VC installation path, you can check it.

Reply:
Hmm, I wonder if we really need such complex way.
Using VSxxxCOMNTOOLS is enough, isn't it?